biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -264,9 +264,20 @@ public  ChemComp getChemComp(String recordName) {
 		} catch (IOException e) {
 
 			logger.error("Could not parse chemical component file "+filename, e);
-			//e.printStackTrace();
 
 		}
+		finally{
+			// Now close it
+			if(inStream!=null){
+				try {
+					inStream.close();
+				} catch (IOException e) {
+					// This would be weird...
+					logger.error("COULD NOT CLOSE CHEMICAL COMPONENT FILE ->>>>> RESOURCE LEAK RESOURCE LEAK. MAYDAY!!!!!!!!! MAYDAY!!!!");
```

Comment:
We can't throw the IOException in there since the method signature (inherited from an interface) does not declare a "throws". Changing that would require some API changes. I agree that would be a better design, where we then would deal with the IOException further down the line.

Note that the actual resource leak happened because `inStream` wasn't being closed at all, whether an IOException happened or not. By putting the `close()` call in the `finally` we make sure it always closes (only in the unlikely case that the closing causes an exception then there could be resources left opened).